Navigation-Examples/Examples/Custom-User-Location.swift

```diff
@@ -0,0 +1,307 @@
+        routes = nil
+    }
+    
+    @objc func performAction(_ sender: Any) {
+        // Set Up the alert controller to switch between different userLocationStyle.
+        let alertController = UIAlertController(title: "Choose UserLocationStyle",
+                                                message: "Select the user location style",
+                                                preferredStyle: .actionSheet)
+        
+        typealias ActionHandler = (UIAlertAction) -> Void
+        
+        let courseView: ActionHandler = { _ in self.setupCourseView() }
+        let puck2D: ActionHandler = { _ in self.setupPuck2D() }
+        let puck3D: ActionHandler = { _ in self.setupPuck3D() }
+        let cancel: ActionHandler = { _ in self.startButtonHighlighted = false }
+        
+        let actionPayloads: [(String, UIAlertAction.Style, ActionHandler?)] = [
+            ("Default Course View", .default, courseView),
+            ("2D Arrow Puck", .default, puck2D),
+            ("3D Car Puck", .default, puck3D),
+            ("Cancel", .cancel, cancel)
+        ]
+        
+        actionPayloads
+            .map { payload in UIAlertAction(title: payload.0, style: payload.1, handler: payload.2) }
+            .forEach(alertController.addAction(_:))
+        
+        if let popoverController = alertController.popoverPresentationController {
+            popoverController.sourceView = self.startButton
+            popoverController.sourceRect = self.startButton.bounds
+        }
+        
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    func setupCourseView() {
+        // Given configuration to the `UserLocationStyle.courseView` through the customizing of `UserPuckCourseView`. Both `UserPuckCourseView` and `UserHaloView` are subclassable.
+        // By default `NavigationMapView.userLocationStyle` property is set to `UserLocationStyle.courseView(_:)`.
+        presentNavigationViewController()
+    }
+    
+    func setupPuck2D() {
+        // It's optional to set up `Puck2DConfiguration` to the `UserLocationStyle.puck2D`. Otherwise the defualt configutaion for the `UserLocationStyle.puck2D` is `Puck2DConfiguration()`.
+        var puck2DConfiguration = Puck2DConfiguration()
+        if #available(iOS 13.0, *) {
+            puck2DConfiguration.topImage = UIImage(systemName: "arrow.up")
+            puck2DConfiguration.scale = .constant(2.0)
+        }
+        
+        let userLocationStyle = UserLocationStyle.puck2D(configuration: puck2DConfiguration)
+        presentNavigationViewController(userLocationStyle)
+    }
+    
+    func setupPuck3D() {
+        // It's required to provide a `Puck3DConfiguration` to the `UserLocationStyle.puck3D`, a `gltf` 3D asset will be used as the `Model` source.
+        // The model source is from NASA's curiosity(clean) in https://github.com/nasa/NASA-3D-Resources/blob/master/3D%20Models/Curiosity%20(Clean)/MSL_clean.blend
```

Comment:
Let’s link to the permalink just in case the file gets renamed or moved in the future:

```suggestion
        // NASA's Curiosity rover from https://github.com/nasa/NASA-3D-Resources/blob/463f2b2941e3e95acfcbb12f783070e9d92ee267/3D%20Models/Curiosity%20(Clean)/MSL_clean.blend
```

Reply:
Thanks for the suggestion. I added a gif example of the Curiosity in SF ☝️